crates/sui-types/src/messages.rs

```diff
@@ -752,14 +752,20 @@ impl Display for TransactionKind {
 }
 
 #[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
-pub struct TransactionData {
-    pub kind: TransactionKind,
-    sender: SuiAddress,
-    gas_payment: ObjectRef,
+pub struct GasData {
+    pub gas_payment: ObjectRef,
+    pub gas_owner: SuiAddress,
```

Comment:
I'm curious...do we even need owner here? given an object can't we lookup the owner of said object. I may be misremembering though...

Reply:
You are not wrong. But if we want to retain the ability to verify a message (SenderSignedData or its envelopes), we'd need to know all the signers without checking the ownership yet. In our codebase today, we would first check a tx's sig, and then check input objects, including ownership. It's useful to stick with that behavior.

Reply:
but there's nothing that stops me from building a txn, using an "owner" field i have the keys for and supply objects that I do not own. Some tooling will be able to verify the message (incorrectly) and the the txn will still fail to execute when the validators do the actual verification on their side. So does this just end up being wasted bits?

Reply:
we could park this for now according to the offline discussion